drivers/aks/aks_driver.go

```diff
@@ -114,6 +114,22 @@ type state struct {
 	// Tags tag the cluster. [optional]
 	Tags map[string]string `json:"tags,omitempty"`
 
+	/**
+	Auto-scaling related properties
```

Comment:
Does it make sense to have a struct here?

Reply:
Sorry, could you elaborate on what you’re suggesting?

Reply:
If you're suggesting putting EnableAutoScaling, MinCount,MaxCount, AgentPoolType, AvailabilityZones & LoadBalancerSku into a sub-struct; I could but was keeping close to the existing AKS & implementations - though I do note that GKE does use some `*raw` structs.

Note that AgentPoolType, AvailabilityZones & LoadBalancerSku are not necessarily limited to only when auto-scaling is enabled.

I don't feel it needs splitting into a sub-structure - but I'm eager to do what is necessary to get this functionality included in rancher - as the experts/owners, I'm happy for you guys to make a call on this.

Your time reviewing is much appreciated & I know it's often a thankless & time-consuming task - Thanks!